packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -15,10 +15,40 @@ import {
 
 export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
 	public parsedSubcommandMappings: SubcommandMappingArray;
+	private caseInsensitiveSubCommands = false;
 
 	public constructor(context: PieceContext, options: O) {
 		super(context, options);
 		this.parsedSubcommandMappings = options.subcommands ?? [];
+
+		const clientOptions = this.container.client.options;
+		if (clientOptions.caseInsensitiveCommands) {
+			// Because slash commands must be lowercase anyways, blanketing all commands to lowercase is safe.
+			this.parsedSubcommandMappings.forEach((cmd) => cmd.name.toLowerCase());
+			this.caseInsensitiveSubCommands = true;
+		}
+
+		if (options.generateDashLessAliases) {
+			for (const mapping of this.parsedSubcommandMappings) {
+				if (!('messageRun' in mapping)) {
+					continue;
+				}
+
+				const dashLessAliases: string[] = [];
+
+				if (this.name.includes('-')) {
+					dashLessAliases.push(mapping.name.replace(/-/g, ''));
+				}
+
+				// For every dashless alias, push a new subcommand with the lowercase alias
+				dashLessAliases.forEach((alias) => {
+					this.parsedSubcommandMappings.push({
+						...mapping,
+						name: alias
+					});
+				});
```

Comment:
```suggestion
				// For every dashless alias, push a new subcommand with the lowercase alias
				for (const alias of dashLessAliases) {
					this.parsedSubcommandMappings.push({
						...mapping,
						name: alias
					});
				}
```

`for...of` is generally faster than `forEach`

Reply:
today I learned